src/widgets/helper/ChannelView.cpp

```diff
@@ -917,8 +917,24 @@ void ChannelView::drawMessages(QPainter &painter)
             isLastMessage = this->lastReadMessage_.get() == layout;
         }
 
+        bool isPastMidnight = false;
+        if (i != messagesSnapshot.size() - 1)
+        {
+            auto elements = &messagesSnapshot[i]->getMessage()->elements;
+            auto elementsAfter =
+                &messagesSnapshot[i + 1]->getMessage()->elements;
+
+            for (int j = 0; j < elements->size(); j++)
+                if ((*elements)[j]->getTime() != QTime(0, 0))
+                    for (int k = 0; k < elementsAfter->size(); k++)
+                        if ((*elementsAfter)[k]->getTime() != QTime(0, 0))
+                            isPastMidnight =
+                                (*elements)[j]->getTime().hour() >
+                                (*elementsAfter)[k]->getTime().hour();
```

Comment:
```suggestion
                            isPastMidnight = (*elements)[j]->getTime() >
                                             (*elementsAfter)[k]->getTime();
```

Reply:
Reason I wanted the hour was because I was unsure if I'd ever get a race condition where one message gets sent before the other one. But that case might not happen.